pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -183,7 +183,10 @@ cpdef convert_to_timedelta64(object ts, object unit):
             ts = cast_from_unit(ts, unit)
             ts = np.timedelta64(ts)
     elif is_string_object(ts):
-        ts = np.timedelta64(parse_timedelta_string(ts))
+        if len(ts) > 0 and ts[0] == 'P':
```

Comment:
Hmm what problem are you trying to solve with this addition? ISO format support should have been added back in #19191 as a precursor to this PR

Reply:
`pd.to_timedelta` was not converting iso-formated strings
In `pd.Timedelta` constructor `parse_iso_format_string` function is used:
```python
if len(value) > 0 and value[0] == 'P':
    value = parse_iso_format_string(value)
else:
    value = parse_timedelta_string(value)
```
while in `pd.to_timedelta` `array_to_timedelta64` is used. Which is not using `parse_iso_format_string`, actually there're only two occurrences of this function in `timedeltas.pyx`, its definition and inside `pd.Timedelta`.
Also, why have `parse_iso_format_string` and `parse_timedelta_string` simultaneously?

Reply:
Ah OK I see. If that's the case then it would be better to do a separate PR as a pre-cursor to this one. That way we can communicate the enhancement and ensure appropriate testing.

Since you started on that, do you want to open an issue about `pd.to_timedelta` not accepting the ISO duration and send a separate PR to fix accordingly? Can come back to this one after the fact

Reply:
OK, I'll open an issue

Reply:
I've opened an issue  #21877